core/sql/optimizer/ValueDesc.cpp

```diff
@@ -529,14 +529,13 @@ NABoolean ValueId::isAddedColumnWithNonNullDefault() const{
   default:
       break;
   }
-  if (nac && nac->isAddedColumn() && nac->getDefaultValue())
+  if (nac &&  nac->getDefaultValue() && strcmp(nac->getDefaultValue(),"NULL") != 0)
```

Comment:
Agree with Selva that checking the enum is safer and more maintainable. Maybe you can make this change in a future check-in, so you don't have to rerun the regressions just for that?